kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaInputOperator.java

```diff
@@ -0,0 +1,512 @@
+          partitioner = new OneToManyPartitioner(clusters, topics, this);
+          break;
+        case ONE_TO_MANY_HEURISTIC:
+          throw new UnsupportedOperationException("Not implemented yet");
+        default:
+          throw new RuntimeException("Invalid strategy");
+      }
+      logger.info("Actual Partitioner is {}", partitioner.getClass());
+    }
+
+  }
+
+  @Override
+  public Response processStats(BatchedOperatorStats batchedOperatorStats)
+  {
+    long t = System.currentTimeMillis();
+    if (repartitionInterval < 0 || repartitionCheckInterval < 0 ||
+        t - lastCheckTime < repartitionCheckInterval || t - lastRepartitionTime < repartitionInterval) {
+      // return false if it's within repartitionCheckInterval since last time it check the stats
+      Response response = new Response();
+      response.repartitionRequired = false;
+      return response;
+    }
+
+    try {
+      logger.debug("Process stats");
+      initPartitioner();
+      return partitioner.processStats(batchedOperatorStats);
+    } finally {
+      lastCheckTime = System.currentTimeMillis();
+    }
+  }
+
+  @Override
+  public Collection<Partition<AbstractKafkaInputOperator>> definePartitions(
+      Collection<Partition<AbstractKafkaInputOperator>> collection, PartitioningContext partitioningContext)
+  {
+    logger.debug("Define partitions");
+    initPartitioner();
+    return partitioner.definePartitions(collection, partitioningContext);
+  }
+
+  @Override
+  public void partitioned(Map<Integer, Partition<AbstractKafkaInputOperator>> map)
+  {
+    // update the last repartition time
+    lastRepartitionTime = System.currentTimeMillis();
+    initPartitioner();
+    partitioner.partitioned(map);
+  }
+
+  /**
+   *
+   * A callback from consumer after it commits the offset
+   * @param map
+   * @param e
+   */
+  public void onComplete(Map<TopicPartition, OffsetAndMetadata> map, Exception e)
+  {
+    if (logger.isDebugEnabled()) {
+      logger.debug("Commit offsets complete {} ", Joiner.on(';').withKeyValueSeparator("=").join(map));
+    }
+    if (e != null) {
+      logger.warn("Exceptions in committing offsets {} : {} ",
+        Joiner.on(';').withKeyValueSeparator("=").join(map), e);
+    }
+  }
+
+
+
+  //---------------------------------------------setters and getters----------------------------------------
+  public void setInitialPartitionCount(int partitionCount)
+  {
+    this.initialPartitionCount = partitionCount;
+  }
+
+  public int getInitialPartitionCount()
+  {
+    return initialPartitionCount;
+  }
+
+  public void setClusters(String clusters)
+  {
+    this.clusters = clusters.split(";");
+  }
+
+  public String getClusters()
+  {
+    return Joiner.on(';').join(clusters);
+  }
+
+  public void setTopics(String... topics)
+  {
+    this.topics = topics;
+  }
+
+  public String[] getTopics()
+  {
+    return topics;
+  }
+
+  public void setConsumerWrapper(KafkaConsumerWrapper consumerWrapper)
```

Comment:
There are 2 extra properties you can set on the wrapper, but I think it's better to move them out to Operator so users can easily set the value and we can make this wrapper object transient